lang/cs/Org.Apache.REEF.Common/Runtime/Evaluator/Context/ContextRuntime.cs

```diff
@@ -226,17 +226,30 @@ public void StartTask(TaskConfiguration taskConfiguration, string contextId, Hea
                     var e = new InvalidOperationException("Attempting to instantiate a child context on a context that is not the topmost active context.");
                     Utilities.Diagnostics.Exceptions.Throw(e, LOGGER);
                 }
+
+                var taskInjector = _contextInjector.ForkInjector(taskConfiguration);
+                var taskId = string.Empty;
+                try
+                {
+                    taskId = taskInjector.GetNamedInstance<TaskConfigurationOptions.Identifier, string>();
```

Comment:
GetNamedInstance is still a hack. You should use task configuration to inject a TaskConfiguration object and make  TaskConfigurationOptions.Identifier as a named parameter of the class.